encode.js

```diff
@@ -26,7 +26,8 @@ function encode(obj, pbf) {
     e = Math.min(e, maxPrecision);
     var precision = Math.ceil(Math.log(e) / Math.LN10);
 
-    var keysArr = Object.keys(keys);
+    var keysArr = Object.keys(keys).sort(
+        function (a, b) { return keys[a] - keys[b]; });
```

Comment:
This compare function doesn't look right...  `a` and `b` will be the key values so `keys[a]` and `keys[b]` will be `undefined`. But since keys are strings, you can just do `Object.keys(keys).sort();`.

Reply:
The intention is sort the key names list by their associate array values (the integer index positions). `Object.keys(keys).sort();` sorts by key name value, not the associate array value. Since `Object.keys(keys)` produces a list of the key names, `keys[a]` and `keys[b]` are defined since `a` and `b` represent elements of this list.

As simple example:
```
keys = {'z' : 3, 'g' : 1, 'a' : 2}

console.log(Object.keys(keys).sort());

console.log(Object.keys(keys).sort(
    function (a, b) { return keys[a] - keys[b]; }))
```

produces:
```
[ 'a', 'g', 'z' ]
[ 'g', 'a', 'z' ]
```

Reply:
Oh, I see. Makes sense.

Reply:
Alternatively, using an "ordered dictionary" data structure for `keys` would be more intuitive. It seems `Map` introduced in ES6 is exactly that. See http://www.jstips.co/en/javascript/map-to-the-rescue-adding-order-to-object-properties/

Perhaps this is a better approach?

Reply:
@mourner I've implemented the Map based approach in #92 for comparison